source/loader/layers/ur_proxy_layer.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+ *
+ * Copyright (C) 2023 Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * @file ur_proxy_layer.h
+ *
+ */
+#ifndef UR_PROXY_LAYER_H
+#define UR_PROXY_LAYER_H 1
+
+#include "ur_ddi.h"
+#include "ur_util.hpp"
+
+///////////////////////////////////////////////////////////////////////////////
+class __urdlllocal proxy_layer_context_t
+{
+    public:
+        ur_api_version_t version = UR_API_VERSION_0_9;
+
+        virtual bool isEnabled() = 0;
+        virtual ur_result_t init(ur_dditable_t *dditable) = 0;
```

Comment:
For now, I think this is fine, but it would be best if we created a generic `init` function to avoid duplication (code for the tracing layer will be nearly identical) and, in the future, allow the layers to be dynamically loaded just as implementations of UR spec, and not some custom API.

Reply:
I think this could be done as an implementation of a "dynamic" proxy layer that loads the appropriate symbols.